lib/foreman_remote_execution_core/polling_script_runner.rb

```diff
@@ -35,18 +52,36 @@ def trigger(*args)
 
     def refresh
       err = output = nil
-      with_retries do
+      begin
         _, output, err = run_sync("#{@user_method.cli_command_prefix} #{@retrieval_script}")
+      rescue => e
+        @logger.info("Error while connecting to the remote host on refresh: #{e.message}")
       end
+      return if output.nil? || output.empty?
       lines = output.lines
       result = lines.shift.match(/^DONE (\d+)?/)
       publish_data(lines.join, 'stdout') unless lines.empty?
       publish_data(err, 'stderr') unless err.empty?
       if result
         exitcode = result[1] || 0
         publish_exit_status(exitcode.to_i)
-        run_sync("rm -rf \"#{remote_command_dir}\"") if @cleanup_working_dirs
+        cleanup
+      end
+    ensure
+      destroy_session
+    end
+
+    def external_event(event)
+      data = event.data
+      return no_update unless data['manual_mode']
+      continuous_output = ForemanTasksCore::ContinuousOutput.new
+      if data.key?('output')
+        lines = Base64.decode64(data['output']).sub(/\A(RUNNING|DONE).*\n/, '')
+        continuous_output.add_output(lines, 'stdout')
       end
+      cleanup if data['exit_code']
+      new_update(continuous_output, data['exit_code'])
```

Comment:
The second argument needs to be casted to an integer, otherwise the job would fail by comparing `'0'` to `0`

Reply:
Were you able to reproduce this? What was the scenario?

Reply:
The exit code sent in the callback is either `nil` or string representation of the number[1]. If the exit code is obtained by smart proxy checking on the host, it works fine as different codepath is used.

[1] - https://github.com/theforeman/foreman_remote_execution/pull/413/files#diff-df5650054157c40b5fc9ca52bf08f2cfR33